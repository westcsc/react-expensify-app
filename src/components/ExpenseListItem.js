import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => {

    console.log(id, description, amount,createdAt);

    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
                
        <p>{amount} - { moment(createdAt).format('MMM Do, YYYY') }</p>
        </div>
    )
}
 
export default ExpenseListItem;