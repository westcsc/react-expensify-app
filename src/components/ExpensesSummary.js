import React from 'react';
import { connect } from 'react-redux';
import  getTotalExpenses from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <p>Viewing {props.expenses.length === 1 ?
        '1 expense ' : 
        `${props.expenses.length} expenses`} totalling {numeral(getTotalExpenses(props.expenses)/100).format('$0,0.00')}
    </p>
);

const mapStateToProps = (state, props) => {
    return {
        expenses: state.expenses
    };
};

export default connect(mapStateToProps)(ExpensesSummary);