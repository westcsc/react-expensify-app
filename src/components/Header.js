import React from "react";
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Go to Home Page</NavLink>
        <NavLink to="/create" activeClassName="is-active">Go to Create Expense page</NavLink>
        <NavLink to="/help" activeClassName="is-active">Go to Help Page</NavLink>
    </header>
);

export default Header;