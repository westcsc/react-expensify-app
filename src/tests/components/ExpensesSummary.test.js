import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={30000} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={36000} />);
    expect(wrapper).toMatchSnapshot();
});