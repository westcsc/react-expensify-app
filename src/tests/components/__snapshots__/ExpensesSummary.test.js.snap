// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing  
    1
     
    expense
     totalling 
    $300.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing  
    3
     
    expenses
     totalling 
    $360.00
  </h1>
</div>
`;
